src/scenarios/shared/startup.py

```diff
@@ -113,18 +114,24 @@ def runtests(self, traits: TestTraits):
         try:
             RunCommand(startup_args, verbose=True).run()
         except CalledProcessError:
+            getLogger().info("Run failure registered")
             if runninginlab():
                 upload_container = 'failedresults'
                 reportjson = os.path.join(
                     TRACEDIR,
                     'FailureReporter', 
                     'failure-report.json')
+                if iswin:
+                    executable = 'FailureReporting.exe'
+                else:
+                    executable = 'FailureReporting'
```

Comment:
Take a look at this example:
https://github.com/dotnet/performance/blob/0c988339004e22574f83c36ea12179a1c26aafc0/src/scenarios/shared/startup.py#L54

I think we should do something similar here. You can port `extension()` as well.